File: 241004/최소 점프 횟수/min-num-of-jumps.cpp
#include <iostream>
#include <vector> 
using namespace std;


int n;
vector<int> arr; 

int result = n+1; 

void find_path(int idx, int cnt){


    if (idx >= n-1){
        result = min(result, cnt);
        return;
    }

    // if (arr[idx] == 0) {
    // return;  // 더 이상 전진 불가, 이 경로는 실패.
    // }
    for(int i = 1; i < arr[idx]+1; i++){
        find_path(idx+i, cnt+1);
    }
}

int main() {
    // 여기에 코드를 작성해주세요.
    cin >> n; 
    result = n+1;

    for(int i = 0; i < n; i++){
        int a; 
        cin >> a; 
        arr.push_back(a);
    }

    find_path(0, 0);
    // for (auto a:arr)
    //     cout << a << " ";
    
    if (result == n+1){
        result = -1; 
    }
    cout << result;


    return 0;
}

File: 241004/최소 점프 횟수/min-num-of-jumps.py
n = int(input())
arr = list(map(int, input().split()))

import sys
result = n+1 

def find_path(index, cnt):
    global result 

    if index >= (n-1):

        result = min(result, cnt)
        return

    if arr[index] == 0:
        return 

    for i in range(1, arr[index]+1):
        # print(index, cnt)
        find_path(index+i, cnt+1)
    
    
find_path(0, 0)



if result == (n+1):
    print(-1)
else:
    print(result)